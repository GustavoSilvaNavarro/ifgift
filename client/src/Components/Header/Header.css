.header {
  display: flex;
  background-color: #fff;
  height: 10vh;
  width: 100%;
  justify-content: space-between;
  margin: 0;
  box-shadow: 2px 0px 4px 1px rgba(17, 138, 178, 0.5);
}

.profile-btn {
  font-size: 2rem;
  background: none;
  border: none;
  text-decoration: none;
  font-weight: bolder;
  stroke: #757575;
  stroke-width: 0.4;
  color: #757575;
  margin: 5px 10px;
  padding: 10px 15px;
}

.profile-btn:hover {
  color: #EF476F;
  stroke: #EF476F;
  background-color: #06D6A0;
  border-radius: 6px;
}

.btn-name {
  display: flex;
  font-size: 0.8rem;
  margin-top: 2px;
  justify-content: center;
}