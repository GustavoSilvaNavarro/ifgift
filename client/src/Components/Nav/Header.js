import './Nav.css';
import { BrowserRouter, Link } from 'react-router-dom';
import { Nav } from "rsuite";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faIdCard, faGift } from '@fortawesome/free-solid-svg-icons';

library.add(faIdCard, faGift)


function Header() {
  return (
    <>
      <nav className="header">
          <div as={Link} to="/home" className="nav-btn"><FontAwesomeIcon icon="fa-solid fa-gift" /><span className="btn-name">home</span></div>
          <div as={Link} to="/profile" className="profile-btn"><FontAwesomeIcon icon="fa-solid fa-id-card" /><span className="btn-name">profile</span></div>
      </nav>
    </>
  );
}

export default Header;