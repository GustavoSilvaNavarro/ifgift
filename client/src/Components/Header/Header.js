import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faIdCard, faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
import './Header.css';

library.add(faIdCard, faMagnifyingGlass)


function Header() {
  return (
    <nav className="header">
        {/* <Link to="/search" className="profile-btn">
          <FontAwesomeIcon icon="fa-solid fa-magnifying-glass" />
          <span className="btn-name">search</span>
        </Link> */}

        <img
        src="https://static.wixstatic.com/media/3dbed1_4a14f7c810ff4d40ae1be96c5c171caa~mv2.png/v1/fill/w_541,h_168,al_c,q_85,usm_0.66_1.00_0.01,enc_auto/ifGift.png" 
        className ="header-logo"
        alt="ifGift logo" />
        
        {/* <Link to="/" className="profile-btn">
          <FontAwesomeIcon icon="fa-solid fa-id-card" />
          <span className="btn-name">profile</span>
        </Link> */}
    </nav>
  );
}

export default Header;