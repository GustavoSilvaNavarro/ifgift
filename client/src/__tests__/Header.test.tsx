/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { Header } from '../Components/Header/Header';

describe('Header Component Tests', () => {
  test('Header Component should match Snapshot', () => {
    const headerTree = create(<Header />).toJSON();

    expect(headerTree).toMatchSnapshot();
  });

  test('Logo Image should be rendered in the document', () => {
    render(<Header />);

    const img = screen.getByRole('img') as HTMLImageElement;

    expect(img.src).toContain(
      'https://static.wixstatic.com/media/3dbed1_4a14f7c810ff4d40ae1be96c5c171caa~mv2.png/v1/fill/w_541,h_168,al_c,q_85,usm_0.66_1.00_0.01,enc_auto/ifGift.png'
    );
  });
});
