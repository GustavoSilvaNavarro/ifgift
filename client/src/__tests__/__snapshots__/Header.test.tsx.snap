// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Tests Header Component should match Snapshot 1`] = `
<nav
  className="header"
>
  <img
    alt="ifGift logo"
    className="header-logo"
    src="https://static.wixstatic.com/media/3dbed1_4a14f7c810ff4d40ae1be96c5c171caa~mv2.png/v1/fill/w_541,h_168,al_c,q_85,usm_0.66_1.00_0.01,enc_auto/ifGift.png"
  />
</nav>
`;
